[23기 차예은]Modified: 구매페이지 수정사항 반영해서 수정

Coderchacha opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
-구매페이지 수정사항 반영해서 수정

## 수정 사항들 자세한 내용
-구매페이지 margin 단축속성 사용해서 수정했습니다.
-구매페이지 return 문이랑 위쪽 변수선언부 사이에 공백 주었습니다.
-API 주소를 별도 파일에 저장해두고 import해오는 방식으로 수정했습니다.

## 기타 질문 및 특이 사항
N/A

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Coderchacha committed:
Add:
-구매페이지 배송정보 기능 추가
-구매페이지 결제 예정 금액 기능 추가

yeonuk-hwang commented on src/pages/Payment/Payment.js:
```diff
@@ -8,31 +8,85 @@ class Payment extends React.Component {
   }
 
+  getTotalPrice = cartItems => {
+    let totalPrice = 0;
+    let totalDiscount = 0;
+    for (let cartItem of cartItems) {
+      const { quantity } = cartItem;
+      const productPrice = cartItem.information[0].product_price;
+      const productDiscount = cartItem.information[0].product_discount;
+
+      totalPrice += quantity * productPrice;
+      totalDiscount += quantity * productDiscount;
+    }
+    return [totalPrice, totalDiscount];
```
이부분 나중에 Array.reduce method 쓰는 방식으로 한번 시도해보셔도 좋을 것 같습니다!
지금 당장 수정할지 말지는 예은님 판단하에 진행해주세요!
이렇게 배열을 순회하면서 어떤 동작을 처리할 때 for문으로 절차적으로 처리해줘도 되지만 그보다 Array의 내장 메소드인 map, filter, reduce, forEach등을 사용해서 처리하면 좀 더 가독성도 좋고 코드의 의도가 잘 드러납니다!(조금 더 안정적이기도 하고요!)

Coderchacha replied:
네 연욱님! 제가 아직 reduce메소드가 익숙하지 않아서 주말에 공부 후에 수정해보겠습니다!

yeonuk-hwang commented on src/pages/Payment/Payment.js:
```diff
@@ -8,31 +8,85 @@ class Payment extends React.Component {
+  };
+
+  handleSubmit = () => {
+    const token = localStorage.getItem('TOKKEN');
```
이부분은 나현님 PR에 TOKKEN key 상수화 해서 사용하라고 코멘트 남겼는데 나현님이랑 얘기해보시고, 팀원 전체적으로 공유해서 적용해주시면 좋아보입니다!

Coderchacha replied:
네 연욱님! 오후에 팀원분들과 이야기 나눠 본 후에 바꿔보겠습니다!

yeonuk-hwang commented on src/pages/Payment/Payment.js:
```diff
@@ -8,31 +8,85 @@ class Payment extends React.Component {
+
+  handleSubmit = () => {
+    const token = localStorage.getItem('TOKKEN');
+    fetch('http://10.58.2.84:8000/orders', {
```
API 주소는 src/config.js 등 별도 파일에 저장해두고 사용하면 좋습니다!

Coderchacha replied:
API 주소를 별도 파일에 저장해두고 import해오는 방식으로 수정했습니다!

yeonuk-hwang commented on src/pages/Payment/Payment.js:
```diff
@@ -8,31 +8,85 @@ class Payment extends React.Component {
   render() {
+    const { cartItem } = this.state;
+    const [totalPrice, totalDiscount] = this.getTotalPrice(cartItem);
     return (
```
return 문이랑 위쪽 변수선언부는 한줄 공백 주면 좋겠네요!

Coderchacha replied:
네 공백 주었습니다!

yeonuk-hwang commented on src/pages/Payment/Payment.scss:
```diff
@@ -2,6 +2,9 @@
   flex-direction: column;
   align-items: center;
+  justify-content: center;
+  margin-left: auto;
+  margin-right: auto;
```
margin 단축속성 사용해보세요~

Coderchacha replied:
margin 단축속성으로 수정했습니다.

yeonuk-hwang reviewed (commented):
리뷰 확인 및 반영해주세요~ 그리고 PR 본문에 수정사항들 자세한 내용도 기입해주세요!
PR 본문을 보고 대략적인 수정사항을 파악할 수 있게요!

Coderchacha committed:
Modify:
-구매페이지 marign 단축속성 사용해서 수정
-API주소를 별도 파일에 저장해두고 import해오는 방식으로 수정
-return 문이랑 위쪽 변수선언부 사이에 공백

yeonuk-hwang commented:
Conflict 해결해주세요~

Coderchacha committed:
branch 이동용 커밋

Coderchacha committed:
Merge branch 'main' into feature/paymentFeature

Coderchacha commented:
연욱님 컨플릭트 해결했습니다!

Coderchacha committed:
Modify:  TOKKEN key 상수화
